### DISCUSSION TITLE
VM Exception while processing transaction: revert

### DISCUSSION THREAD

**Discussion by adivated** [tryethernal/ethernal#72] [General]
Just deployed a simple ERC20 contract to ganache. Imported ABI, but all tests against the contract just return: "VM Exception while processing transaction: revert"
even simple calls like name, symbol, totalSupply queries (no parameters) return that same error. Any ideas? ethernal is picking up the tx and contracts ok, so it's syncing.

**Comment by adivated** [comment]
well, not sure. I just redeployed the contract again and it works fine, so 🤷‍♂️

**Reply by antoinedc** [reply]
That can happen if the chain is restarted between the deployment of the contract and the call. Do you think this is what might have happened?

**Reply by adivated** [reply]
No the chain wasn’t restarted.  I realize ganache resets chain data though so that is something to watch for. 👍

**Reply by antoinedc** [reply]
While the chain is running? When does that happen?

**Reply by adivated** [reply]
No when GUI is restarted unless you persist the db

**Reply by antoinedc** [reply]
Ah I see, yes, it resets on chain data, but the Ethernal dashboard can't detect that (yet), so data in there might be outdated.
In those cases, the best thing to do is to go in "Settings" > "Reset workspace", to start fresh!
